report-jtreg-lib/src/main/java/io/jenkins/plugins/report/jtreg/ConfigFinder.java

```diff
@@ -93,4 +99,53 @@ private static String findInProperties(File configFile, String key) {
             return null;
         }
     }
+
+    private static String findInJson(File configFile, String jsonQuery) {
+        String[] strings = jsonQuery.split("\\."); // split the parsing string on dots
+        try {
+            InputStream stream = new FileInputStream(configFile);
+            String jsonString = IOUtils.toString(stream, StandardCharsets.UTF_8);
+            JsonObject current = new Gson().fromJson(jsonString, JsonObject.class);
+            JsonElement je = null;
+
+            // go through all the parts (separated from the json query by dots)
+            for (String part : strings) {
+                if (je != null) {
+                    current = je.getAsJsonObject();
+                }
+
+                if (part.equals("$") && Arrays.asList(strings).indexOf(part) == 0) {
+                    // if the first is $, ignore it
+                    continue;
+                } else if (part.matches(".*\\[[0-9]*]")) {
+                    // if there is an array in the json query, get the right element
+                    String[] split = part.split("\\[");
+                    int num = Integer.parseInt(split[1].split("]")[0]);
+
+                    // check if it is not null
+                    if (current.get(split[0]) != null) {
+                        je = current.get(split[0]).getAsJsonArray().get(num);
+                    } else {
+                        return null;
+                    }
+                } else {
+                    // get the corresponding part of the json
+                    je = current.get(part);
+                }
+            }
+
+            if (je == null) {
+                return null;
+            } else {
+                try {
+                    // try to get the value as string
+                    return je.getAsString();
+                } catch (Exception e) {
+                    return null;
+                }
+            }
+        } catch (IOException e) {
+            return null;
```

Comment:
aargh those consumed exceptions of yours.... if nothing else, at elast log it.
In thsi case, where method is cosnumer facing, the suer muist be notifed waht went wrong and why....